### PR TITLE
[BugFix] Fix materialized view rewrite bug of agg with count(1)

### LINKED ISSUE #12702: Aggregate materialized view rewrite fails
### Steps to reproduce the behavior (Required)
origin sync materialized view has this bug. Refer to SR-12978

### Expected behavior (Required)
query successfully

### Real behavior (Required)
query failed
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
main

### PR DESCRIPTION
**Pull request by ABingHuang**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #12702 

## Problem Summary(Required) ：

Fix materialized view rewrite bug of agg with count(1)

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by ABingHuang** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/mv/MaterializedViewRewriter.java [hunk 1]
```diff
@@ -119,6 +119,9 @@ public OptExpression visitLogicalAggregate(OptExpression optExpression,
         Map<ColumnRefOperator, CallOperator> newAggMap = new HashMap<>(aggregationOperator.getAggregations());
         for (Map.Entry<ColumnRefOperator, CallOperator> kv : aggregationOperator.getAggregations().entrySet()) {
             String functionName = kv.getValue().getFnName();
+            if (kv.getValue().getUsedColumns().isEmpty()) {
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/mv/MaterializedViewRewriter.java:122]: you should change the checkAggregationFunction logic, not here.

**Review by Youngwb — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/mv/MaterializedViewRewriter.java|2|2|100.00%|[]|

**Review by Astralidea — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Comment by ABingHuang** [comment]
run starrocks_admit_test

**Pull request merged**